core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -255,7 +255,7 @@ private static void validateNonAncestor(TableMetadata meta, long snapshotId) {
 
   private static void validateReplacedPartitions(TableMetadata meta, Long parentId,
                                                  PartitionSet replacedPartitions) {
-    if (replacedPartitions != null) {
+    if (replacedPartitions != null && meta.currentSnapshot() != null) {
```

Comment:
Dumb question here: is it safe for `validateCurrentSnapshot` to not check `meta.currentSnapshot() != null` in prior? If `requiredCurrentSnapshotId` is not null and `meta.currentSnapshot()` is null then NPE would occur, but I'm not sure the possibility about it.

Reply:
Looks like we should fix that one as well. I think you should be able to FF to any snapshot if there is no current snapshot. Thanks @HeartSaVioR!